### DISCUSSION TITLE
Regarding input to model

### DISCUSSION THREAD

**Discussion by mahewashabdi** [google/lightweight_mmm#17] [General]
Hi 

The media_data as per the documentation should be taken as impressions/ clicks etc.
In the demo example it is said that media data can be cost spend on that channel. 
Can you clarify what should be taken in the media_data? 
Also, can I use cost spend as the extra features for the input. what other options are there to be taken in consideration  for extra features ?

Can I consider CPM in prices here as I am using impressions in the media_data
![image](https://user-images.githubusercontent.com/63654651/173605826-f62754a0-15e7-4c1f-8819-7861f78b9262.png)

**Comment by mahewashabdi** [comment]
Hi Team,

I am using impressions which is a money unit so as per your comments in the demo example I shouldn't be multiplying it by prices .
But when I am not doing that the sum of optimal budget I am getting is 432 whereas my budget is 10000 but when I multiply it with prices ( I have used CPMs) then the total sum is coming out to be around 10000.

Can you please explain ?

![image](https://user-images.githubusercontent.com/63654651/173623326-443d24d7-dcc2-4495-b219-d0e49b097e15.png)

**Comment by pabloduque0** [comment]
I will try to answer the multiple questions:
`media_data` can be impresions, clicks or spend. Depending on what you use you might need to do certain things different at certain steps of the process. The example notebooks use spend as `media_data`.

For the extra variables you can use anything that is helpful to predict you KPI but it should be done carefully. Best examples of extra variables are promotions, price information, competitors data, ... to name a few.

If you are using impressions your budget should be set manually to whatever you desire and the prices should also be set to an average price you are paying for each of those impressions (or the expected price for the optimization timeframe). If you use CPM, you need to use CPM / 1000 (if we are talking about CPM as Cost per thousand).

Let us know if there are any remaining questions :)

**Reply by mahewashabdi** [reply]
Thanks @pabloduque0
